symengine/tests/cwrapper/test_cwrapper.c

```diff
@@ -452,6 +452,61 @@ void test_constants() {
 
 }
 
+void test_functions() {
+    basic pi;
+    basic ans;
+
+    basic_new_stack(pi);
+    basic_new_stack(ans);
+
+    basic_const_pi(pi);
+
+    char* s;
+    s = basic_str(pi);
+    SYMENGINE_C_ASSERT(strcmp(s, "pi") == 0);
+    basic_str_free(s);
+
+    basic_sin(ans, pi);
+    s = basic_str(ans);
+    SYMENGINE_C_ASSERT(strcmp(s, "0") == 0);
+    basic_str_free(s);
+    basic_free_stack(ans);
+
+    basic_cos(ans, pi);
+    s = basic_str(ans);
+    SYMENGINE_C_ASSERT(strcmp(s, "-1") == 0);
```

Comment:
For automatically simplified answers, it's good to use `basic_eq` instead of string comparisons. String comparisons are better for checking that `sin(x**2)` prints correctly.